pandas/io/excel/_base.py

```diff
@@ -1016,28 +1078,30 @@ def __init__(
                     else:
                         stacklevel = 2
                     warnings.warn(
-                        "The xlrd engine is no longer maintained and is not "
-                        "supported when using pandas with python >= 3.9. However, "
-                        "the engine xlrd will continue to be allowed for the "
-                        "indefinite future. Beginning with pandas 1.2.0, the "
-                        "openpyxl engine will be used if it is installed and the "
-                        "engine argument is not specified. Either install openpyxl "
-                        "or specify engine='xlrd' to silence this warning.",
+                        f"Your version of xlrd is {xlrd_version}. In xlrd >= 2.0, "
+                        f"only the xls format is supported. As a result, the "
+                        f"openpyxl engine will be used if it is installed "
+                        f"and the engine argument is not specified. Either install "
+                        f"openpyxl or specify engine='xlrd' to silence this warning.",
                         FutureWarning,
                         stacklevel=stacklevel,
                     )
                     engine = "xlrd"
-        if engine not in self._engines:
-            raise ValueError(f"Unknown engine: {engine}")
```

Comment:
Why is this removed here? I see it is moved to the top, but you can still end up here with a wrong/unknown engine at this point?

Reply:
If engine is not None, this is checked above; if it is None, it should only be possible to be an engine that is in the dictionary.  As such, I've added this back in as an assert.

Reply:
Before there was not an `else:` assigning an engine in any case (so it could end up being None), but with your changes that should indeed not be the case anymore.